ouroboros-consensus/src/Ouroboros/Consensus/Node/Run/Abstract.hs

```diff
@@ -62,6 +63,18 @@ class (ProtocolLedgerView blk, ApplyTx blk) => RunNode blk where
   nodeHashInfo            :: Proxy blk
                           -> HashInfo (HeaderHash blk)
 
+  -- | The maximum block size in bytes according to the currently adopted
+  -- protocol parameters of the ledger state.
+  nodeMaxBlockSize :: LedgerState blk -> Word32
+
+  -- | The block encoding overhead size in bytes.
+  --
+  -- This encompasses the overhead in bytes for everything that is encoded
+  -- within a block, excluding the actual generalized transactions. Given
+  -- this, along with the 'nodeMaxBlockSize', it is possible to determine the
+  -- amount of generalized transactions that we can include in a block.
+  nodeBlockEncodingOverhead :: Proxy blk -> Word32
```

Comment:
To work correctly for the hard fork combinator (~ sum of Byron and Shelley blocks), this method should be able to tell whether we're before (forge Byron blocks with Byron encoding overhead) or after the hard fork (forge Shelley blocks with Shelley encoding overhead).
Let it take the `LedgerState blk` as argument, as that will allow us to tell whether we're before or after the hard fork.